content/en/docs/concepts/workloads/pods/user-namespaces.md

```diff
@@ -137,20 +153,55 @@ use, see `man 7 user_namespaces`.
 
 ## Set up a node to support user namespaces
 
-It is recommended that the host's files and host's processes use UIDs/GIDs in
-the range of 0-65535.
-
-The kubelet will assign UIDs/GIDs higher than that to pods. Therefore, to
-guarantee as much isolation as possible, the UIDs/GIDs used by the host's files
-and host's processes should be in the range 0-65535.
+By default, the kubelet assumes host's files and host's processes use UIDs/GIDs
+in the range of 0-65535. This is the case in most Linux distributions with the
+default installation. The kubelet will assign UIDs/GIDs higher than that to
+pods, so there is no overlap between UID/GIDs used by host's files and processes
+and the ones used by pods.
 
 Note that this recommendation is important to mitigate the impact of CVEs like
 [CVE-2021-25741][CVE-2021-25741], where a pod can potentially read arbitrary
-files in the hosts. If the UIDs/GIDs of the pod and the host don't overlap, it
+files in the host. If the UIDs/GIDs of the pod and the host don't overlap, it
 is limited what a pod would be able to do: the pod UID/GID won't match the
 host's file owner/group.
 
+The kubelet can use a custom range for pods too. To configure a custom range,
+the node needs to have:
+
+ * A user `kubelet` in the system
+ * The binary `getsubids` installed (part of the [shadow-utils][shadow-utils] package)
+ * A configuration of subordinate UIDs/GIDs for the `kubelet` user (see `man 5
+   subuid` and `man 5 subgid`)
+
+The `kubelet` user will only be used to gather the UID/GIDs range configuration,
+but the user the kubelet is executed as won't be changed by this.
```

Comment:
```suggestion
The kubelet looks for a user ID and group ID range assigned to the local user named `kubelet`;
the kubelet never switches its own real or effective group ID to that user.
```

Reply:
> It is not a local user

If I run `getent passwd | grep kubelet`, do I see an entry? That's the test I'm imagining.

Reply:
> I wouldn't say the kubelet will never switch the user. Other setups, like running the kubelet inside a userns, might at some point do that.

OK, in that case, we need a _different_ wording as 
> The `kubelet` user will only be used to gather the UID/GIDs range configuration, but the user the kubelet is executed as won't be changed by this.

might mislead people. What really happens?

Reply:
> I wouldn't say the kubelet will never switch the user. Other setups, like running the kubelet inside a userns, might at some point do that.

OK,  so in long form, what information do we want the reader to learn from this paragraph around user switching? I'm not confident I know.

Reply:
@sftim 

> If I run `getent passwd | grep kubelet`, do I see an entry? That's the test I'm imagining.

I expect that, yes. But the output of that command is not just local users AFAIK.



> > The `kubelet` user will only be used to gather the UID/GIDs range configuration, but the user the kubelet is executed as won't be changed by this.
> 
> might mislead people. What really happens?

What happens is we check the mappings for the `kubelet`user and use that for the ranges. We don't change anything about the user used to run the kubelet.



> OK, so in long form, what information do we want the reader to learn from this paragraph around user switching? I'm not confident I know.

Exactly what I said in my previous answer.

Reply:
Would this be more concise, but still accurate?

```suggestion
This setting only gathers the UID/GID range configuration and does not change the user executing the `kubelet`.
```

Reply:
Applied that, thanks.